### ISSUE TITLE
Add user interface for multivalued affiliation field next for each contributor

### ISSUE THREAD

**Issue by peetucket** [sul-dlss/happy-heron#3156]
Mapping work that goes along with this in #3099

Design: https://projects.invisionapp.com/share/NP138M0QX7T3#/screens/473166568
Requirements: https://docs.google.com/presentation/d/1FI9yn7vUSeX7Yd5IemrLxnYH2__JlTxNwgP8guATb18

_Background:_
- we want to record contributor affiliations along with names (and ORCIDs)
- recording affiliations, especially with organization identifiers (RoRs) will enrich metadata and discoverability

_Requirement:_
- Contributor entries on the work edit form should include a new affiliation entry box.
- The field is multivalued so needs an "Add another affiliation" button.
- The value(s) are associated with the contributor.

_Notes:_
- Affiliation is not required
- Multiple affiliations are allowed
- Free text entries are allowed; ROR not required but preferred (see [#3100](https://github.com/sul-dlss/happy-heron/issues/3100) for RoR)

There is a separate ticket for how/what we need to store and how it maps to cocina (in #3099).  If questions about what we need to store are needed before the cocina mapping work is done, ask if Arcadia can quickly answer to unblock work.

**Comment by justinlittman** [comment]
Considering how to handle for organization is covered in #3205
